src/dali/plugin/input/rest/binance_com.py

```diff
@@ -0,0 +1,774 @@
+                            notes=(f"{notes + '; ' if notes else ''} Fee for conversion from " f"{conversion_info}"),
+                        )
+                    )
+            transaction_notes = f"Buy side of conversion from " f"{conversion_info}" f"({out_asset} out-transaction unique id: {transaction[_ID]}"
+
+        in_transaction_list.append(
+            InTransaction(
+                plugin=self.__BINANCE_COM,
+                unique_id=unique_id,
+                raw_data=json.dumps(transaction),
+                timestamp=timestamp,
+                asset=in_asset,
+                exchange=self.__BINANCE_COM,
+                holder=self.account_holder,
+                transaction_type=Keyword.BUY.value,
+                spot_price=str(spot_price),
+                crypto_in=str(crypto_in),
+                crypto_fee=(str(crypto_fee) if (crypto_fee is not None) else None),
+                fiat_in_no_fee=fiat_in_no_fee,
+                fiat_in_with_fee=fiat_in_with_fee,
+                fiat_fee=fiat_fee,
+                notes=(f"{notes + '; ' if notes else ''} {transaction_notes}"),
+            )
+        )
+
+    def _process_deposit(self, transaction: Any, in_transaction_list: List[InTransaction], notes: Optional[str] = None) -> None:
+        self.__logger.debug("Deposit: %s", transaction)
+
+        amount: RP2Decimal = RP2Decimal(transaction[_INDICATEDAMOUNT])
+        fiat_in_with_fee: RP2Decimal = RP2Decimal(transaction[_AMOUNT])
+        fee: RP2Decimal = RP2Decimal(transaction[_TOTALFEE])
+        notes = f"{notes + '; ' if notes else ''}{'Fiat Deposit of '}; {transaction[_FIATCURRENCY]}"
+        in_transaction_list.append(
+            InTransaction(
+                plugin=self.__BINANCE_COM,
+                unique_id=transaction[_ORDERNO],
+                raw_data=json.dumps(transaction),
+                timestamp=self._rp2timestamp_from_ms_epoch(transaction[_CREATETIME]),
+                asset=transaction[_FIATCURRENCY],
+                exchange=self.__BINANCE_COM,
+                holder=self.account_holder,
+                transaction_type=Keyword.BUY.value,
+                spot_price="1",
+                crypto_in=str(amount),
+                crypto_fee=None,
+                fiat_in_no_fee=str(fiat_in_with_fee),
+                fiat_in_with_fee=str(amount),
+                fiat_fee=str(fee),
+                notes=notes,
+            )
+        )
+
+    def _process_gain(self, transaction: Any, transaction_type: Keyword, in_transaction_list: List[InTransaction], notes: Optional[str] = None) -> None:
+
+        if transaction_type == Keyword.MINING:
+            amount: RP2Decimal = RP2Decimal(str(transaction[_PROFITAMOUNT]))
+            notes = f"{notes + '; ' if notes else ''}'Mining profit'"
+            in_transaction_list.append(
+                InTransaction(
+                    plugin=self.__BINANCE_COM,
+                    unique_id=(str(transaction[_TIME]) + transaction[_COINNAME]),

```

Comment:
This is clever, but isn't there some unique id in the transaction?

Reply:
Unfortunately no. There is no unique identifier in the docs or in the real world data. Maybe it is something they overlooked?

I opened a forum post on it on their dev forum, but I wouldn't hold my breath.

Reply:
I'm a bit concerned about corner cases: e.g. two transactions denominated in the same coin, with the same timestamp. That would throw off the transaction resolver. Maybe add some more qualifiers: e.g. crypto / fiat amount, crypto / fiat fee: still not bulletproof though... When you say they don't have it, you mean in the native Binance API, right? I'm actually thinking it might be safer to assign `Keyword.UNKNOWN.value`

Reply:
Yes, as you said, these are gains, so they don't need to be resolved (resolving is meant for incomplete intra transactions, which can be represented as intra-intra or in-out): in this case it's a self-contained transaction and so using unknown should be fine.